pf4j/src/main/java/org/pf4j/Extension.java

```diff
@@ -47,4 +47,16 @@
      */
     Class<? extends ExtensionPoint>[] points() default {};
 
+    /**
+     * An array of plugin IDs, that have to be available in order to load this extension.
+     * The {@link AbstractExtensionFinder} won't load this extension, if these plugins are not
+     * available / started at runtime.
+     * <p>
+     * Notice: This feature requires the optional <a href="https://asm.ow2.io/">ASM library</a>
+     * to be available on the applications classpath and has to be explicitly enabled via
+     * {@link AbstractExtensionFinder#setCheckForExtensionDependencies(boolean)}.
+     *
+     * @return plugin IDs, that have to be available in order to load this extension
+     */
+    String[] plugins() default {};
```

Comment:
I don't like this attribute in the signature of Extension annotation. Can you are more detail about the goal of this attribute?

Reply:
Unfortunately this attribute seems necessary to me in order to ignore extensions, that make use of an **unavailable plugin** at runtime. This avoids a possible `NoClassDefFoundError`'s in the `AbstractExtensionFinder`.

I've described my motivation about this attribute more precisely in my first posting below the headline "How extensions are processed for missing optional plugins".